metapool/prep.py

```diff
@@ -0,0 +1,424 @@
+
+    # no matches
+    matches = re.search(qiita_id_re, project_name)
+    if matches is None:
+        return project_name
+    else:
+        # group 1 is the project name
+        return matches[1]
+
+
+def get_run_prefix(run_path, project, sample, lane, pipeline):
+    """For a sample find the run prefix
+
+    Parameters
+    ----------
+    run_path: str
+        Base path for the run
+    project: str
+        Name of the project
+    sample: str
+        Sample name
+    lane: str
+        Lane number
+    pipeline: str
+        The pipeline used to generate the data. Should be one of
+        `atropos-and-bowtie2` or `fastp-and-minimap2`.
+
+    Returns
+    -------
+    str
+        The run prefix of the sequence file in the lane, only if the sequence
+        file is not empty.
+    """
+    base = os.path.join(run_path, project)
+    path = base
+
+    # each pipeline sets up a slightly different directory structure,
+    # importantly fastp-and-minimap2 won't save intermediate files
+    if pipeline == 'atropos-and-bowtie2':
+        qc = os.path.join(base, 'atropos_qc')
+        hf = os.path.join(base, 'filtered_sequences')
+
+        # If both folders exist and have sequence files always prefer the
+        # human-filtered sequences
+        if _exists_and_has_files(qc):
+            path = qc
+            if _exists_and_has_files(hf):
+                path = hf
+    elif pipeline == 'fastp-and-minimap2':
+        qc = os.path.join(base, 'trimmed_sequences')
+        hf = os.path.join(base, 'filtered_sequences')
+
+        if _exists_and_has_files(qc) and _exists_and_has_files(hf):
+            path = hf
+        elif _exists_and_has_files(qc):
+            path = qc
+        elif _exists_and_has_files(hf):
+            path = hf
+        else:
+            path = base
+    else:
+        raise ValueError('Invalid pipeline "%s"' % pipeline)
+
+    results = glob(os.path.join(path, '%s_S*_L*%s_R*.fastq.gz' %
+                                (sample, lane)))
+
+    # at this stage there should only be two files forward and reverse
+    if len(results) == 2:
+        forward, reverse = sorted(results)
+
+        if is_nonempty_gz_file(forward) and is_nonempty_gz_file(reverse):
+
+            f, r = os.path.basename(forward), os.path.basename(reverse)
+            if len(f) != len(r):
+                raise ValueError("Forward and reverse sequences filenames "
+                                 "don't match f:%s r:%s" % (f, r))
+
+            # The first character that's different is the number in R1/R2. We
+            # find this position this way because sometimes filenames are
+            # written as _R1_.. or _R1.trimmed... and splitting on _R1 might
+            # catch some substrings not part of R1/R2.
+            for i in range(len(f)):
+                if f[i] != r[i]:
+                    i -= 2
+                    break
+
+            return f[:i]
+        else:
+            return None
+
+    else:
```

Comment:
should this raise a warning/error/other? I mean having !=2 files is bad, no?

Reply:
Yes, this would be really bad. Added a warning so that we can see a list of all the files that have multiple matches rather than just the first one.